core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -13,146 +13,623 @@
+const voiceArtistSettingsDropDown =
+  'h3.e2e-test-voice-artists-settings-container';
+const rolesSettingsDropDown = 'h3.e2e-test-roles-settings-container';
+const advanceSettingsDropDown = 'h3.e2e-test-advanced-settings-container';
+const explorationControlsSettingsDropDown =
+  'h3.e2e-test-controls-bar-settings-container';
+const mobileSaveDraftButton = 'button.e2e-test-save-changes-for-small-screens';
 const mobilePublishButton = 'button.e2e-test-mobile-publish-button';
+const publishButtonDropDown = 'div.e2e-test-mobile-changes-dropdown';
+const mobileDiscardButton = 'div.e2e-test-mobile-exploration-discard-tab';
+
+let explorationUrlAfterPublished = '';
+export class ExplorationCreator extends BaseUser {
+  async openCreatorDashboardPage(): Promise<void> {
+    await this.goto(creatorDashboardUrl);
+    showMessage('Creator dashboard page is opened successfully.');
+  }
 
-export class ExplorationEditor extends BaseUser {
   /**
-   * Function to navigate to creator dashboard page
+   * This function clicks on new exploration button.
    */
-  async navigateToCreatorDashboardPage(): Promise<void> {
-    await this.page.goto(creatorDashboardPage);
+  async createNewExploration(): Promise<void> {
+    await this.clickOn(createNewExplorationButton);
   }
 
   /**
-   * Function to navigate to exploration editor
+   * This function closes the tutorial popup before a new exploration.
    */
-  async navigateToExplorationEditorPage(): Promise<void> {
-    await this.clickOn(createExplorationButton);
+  async switchToEditorTab(): Promise<void> {
+    await this.clickOn(tutorialTakeMeToEditorButton);
+    showMessage('Tutorial popup closed successfully.');
   }
 
   /**
-   * Function to create exploration with title
-   * @param explorationTitle - title of the exploration
+   * This function clears previous card name and
+   * add a new card name.
    */
-  async createExplorationWithTitle(explorationTitle: string): Promise<void> {
-    await this.page.waitForFunction('document.readyState === "complete"');
-    await this.page.waitForSelector(textStateEditSelector, {
-      visible: true,
-    });
-    await this.clickOn(textStateEditSelector);
-    await this.page.waitForSelector(richTextAreaField, {visible: true});
-    await this.type(richTextAreaField, `${explorationTitle}`);
-    await this.clickOn(saveContentButton);
+  async updateCardName(cardName: string): Promise<void> {
+    await this.clickOn(editCardNameButton);
+    await this.waitForElementToBeClickable(editCardNameBox);
+    await this.page.click(editCardNameBox, {clickCount: 3});
+    await this.page.keyboard.press('Backspace');
+    await this.type(editCardNameBox, cardName);
+    await this.clickOn(introSubmitButton);
+    showMessage('Card name updated successfully.');
+  }
 
+  /**
+   * This function clears previous intro text and
+   * add a new intro text.
+   */
+  async updateExplorationIntroText(introtext: string): Promise<void> {
```

Comment:
intro and text are two different words, Hence I used `updateExplorationIntroText`.

Why it would be `updateExplorationintroText` please elaborate more.

Reply:
done